### PR TITLE
Use the rocksdb mget to improve the performance

### LINKED ISSUE #1441: Hash::MGet: use DB::MultiGet for hash fields
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Motivation

`DB::MultiGet` pushes fields query to RocksDB, thus RocksDB can perform optimizations, such as async io by io uring.

Although this optimization is less than `mget` for strings, it is valuable because different fields may lies in different levels of the LSM Tree, thus async io can take effect.

### Solution

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by zevin02**
fixes: #1441

- Use RocksDB's MultiGet to improve mget performance of hash

### CODE REVIEW

**Review by mapleFU — DISMISSED** [review]
The code LGTM.

By the way, should we add a `Storage::MultiGet` without `ColumnFamily` like `Get`?

**Comment by zevin02** [comment]
I think I can add it

**Review by tisonkun — DISMISSED** [review]

File: src/types/redis_hash.cc [hunk 1]
```diff
@@ -174,16 +174,27 @@ rocksdb::Status Hash::MGet(const Slice &user_key, const std::vector<Slice> &fiel
...
-    values->emplace_back(value);
-    statuses->emplace_back(s);
+  std::vector<rocksdb::PinnableSlice> values_vector;
+  values_vector.resize(keys.size());
+  std::vector<rocksdb::Status> statuses_vector;
+  statuses_vector.resize(keys.size());
+  storage_->MultiGet(read_options, storage_->GetDB()->DefaultColumnFamily(), keys.size(), keys.data(),
+                     values_vector.data(), statuses_vector.data());
+  for (size_t i = 0; i < keys.size(); i++) {
+    if (!statuses_vector[i].ok() && statuses_vector[i].IsNotFound()) return statuses_vector[i];
```
Comment (git-hulk) [inline src/types/redis_hash.cc]:
```suggestion
    if (!statuses_vector[i].ok() && !statuses_vector[i].IsNotFound()) return statuses_vector[i];
```

Comment (infdahai) [reply src/types/redis_hash.cc]: This statement corresponds to #1455

Comment (git-hulk) [reply src/types/redis_hash.cc]: Yes

**Review by git-hulk — APPROVED** [review]

**Review by mapleFU — APPROVED** [review]
Sorry, miss the NotFound part

**Review by infdahai — APPROVED** [review]

**Comment by mapleFU** [comment]
@zevin02 Could you find or add a testcase for https://github.com/apache/incubator-kvrocks/issues/1455 ?

**Comment by zevin02** [comment]
> @zevin02 Could you find or add a testcase for #1455 ? @zevin02 你能为 #1455 找到或添加一个测试用例吗？

What does it mean? Do I need to add a test case for reproducing this issue? I think the PR I just modified already fixed this issue.

**Comment by git-hulk** [comment]
> Sorry, miss the NotFound part

😊 No problem. Great thanks for your review.

**Comment by infdahai** [comment]
> What does it mean? Do I need to add a test case for reproducing this issue? I think the PR I just modified already fixed this issue.

It means you can just write a test to record this process. (Add a test to the hash test)
https://github.com/apache/incubator-kvrocks/blob/fb0e3d42031a6e0691c7a736711d0a01d72bdb1a/tests/gocase/unit/type/hash/hash_test.go#L53

The test is just used to avoid this situation reproducing due to the code that follows.

**Comment by mapleFU** [comment]
```
--- FAIL: TestSlotMigrateSync (47.71s)
    --- FAIL: TestSlotMigrateSync/MIGRATE_-_Migrate_sync_with_(or_without)_all_kinds_of_timeouts (0.01s)
        slotmigrate_test.go:425: 
            	Error Trace:	/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/slotmigrate_test.go:425
            	Error:      	Not equal: 
            	            	expected: string("OK")
            	            	actual  : <nil>(<nil>)
            	Test:       	TestSlotMigrateSync/MIGRATE_-_Migrate_sync_with_(or_without)_all_kinds_of_timeouts
FAIL
exit status 1
```

Is the unittest failed related?

**Comment by zevin02** [comment]
> ```
> --- FAIL: TestSlotMigrateSync (47.71s)
>     --- FAIL: TestSlotMigrateSync/MIGRATE_-_Migrate_sync_with_(or_without)_all_kinds_of_timeouts (0.01s)
>         slotmigrate_test.go:425: 
>             	Error Trace:	/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/integration/slotmigrate/slotmigrate_test.go:425
>             	Error:      	Not equal: 
>             	            	expected: string("OK")
>             	            	actual  : <nil>(<nil>)
>             	Test:       	TestSlotMigrateSync/MIGRATE_-_Migrate_sync_with_(or_without)_all_kinds_of_timeouts
> FAIL
> exit status 1
> ```
> 
> Is the unittest failed related?


I did not make any changes to the relevant code and I don't know why there is an error.

**Comment by infdahai** [comment]
It is related to #1452.

**Comment by zevin02** [comment]
emm,So what should I do

> It is related to #1452.

**Comment by infdahai** [comment]
Sometimes the test works and sometimes it doesn't. So just ignore it for now.

**Comment by mapleFU** [comment]
@zevin02 Patch LGTM. And the hash case is tested. Let's wait committer rerun the unittest tomorrow.

**Comment by zevin02** [comment]
> > What does it mean? Do I need to add a test case for reproducing this issue? I think the PR I just modified already fixed this issue.
> 
> It means you can just write a test to record this process. (Add a test to the hash test)
> 
> https://github.com/apache/incubator-kvrocks/blob/fb0e3d42031a6e0691c7a736711d0a01d72bdb1a/tests/gocase/unit/type/hash/hash_test.go#L53
> 
> The test is just used to avoid this situation reproducing due to the code that follows.

I can add this test case later

**Comment by git-hulk** [comment]
Thanks all, the patch looks good to be merged.

**Pull request merged**